src/SpendPermissionManager.sol

```diff
@@ -488,14 +499,22 @@ contract SpendPermissionManager is EIP712 {
         // transfer tokens from account to recipient
         if (token == NATIVE_TOKEN) {
             _execute({account: account, target: recipient, value: value, data: hex""});
-        } else {
+            return;
+        }
+        // set infinite allowance if not yet set
+        uint256 allowance = IERC20(token).allowance(account, address(this));
+        if (allowance != type(uint256).max) {
             _execute({
                 account: account,
                 target: token,
                 value: 0,
-                data: abi.encodeWithSelector(IERC20.transfer.selector, recipient, value)
+                data: abi.encodeWithSelector(IERC20.approve.selector, address(this), type(uint256).max)
             });
         }
+
+        // use ERC-20 allowance to transfer from account to recipient
+        // safeTransferFrom will revert if transfer fails, regardless of ERC-20 implementation
+        IERC20(token).safeTransferFrom(account, recipient, value);
```

Comment:
by using safe transfer, openzeppelin has already done the work of correctly managing and interpreting the presence or absence of return data.  I tried to wrap this in a try/catch so that we could supply our own error with all of the context we try to offer in `ERC20TransferFailed`, but since `safeTransferFrom` is a library call, the compiler treats it as an internal call and disallows try/catch.  This is probably fine since I was debating whether to leave any original reverts unmodified anyway. i.e. `InsufficientBalance`